src/toil/test/src/checkpointTest.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (C) 2015-2017 Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from toil.job import Job
+from toil.test import ToilTest
+from toil.leader import FailedJobsException
+from toil.jobStores.abstractJobStore import NoSuchFileException
+
+class CheckpointTest(ToilTest):
+    def testCheckpointNotRetried(self):
+        """A checkpoint job with a retryCount of 0 should not be retried."""
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.retryCount = 0
+        # We set the workflow to *succeed* if the checkpointed job is
+        # retried despite the retry count of 0. So, if the workflow
+        # fails, this test succeeds.
+        # This ends up being a bit confusing, but it's the easiest way
```

Comment:
We should clarify here that it is only true that a root job with a retry count of 0 should be tried only once. The behavior of a sequence with length n of successive checkpoint jobs, each with a retry count of 1 and each a child of the previous job in the sequence is that the last job in the sequence will potentially be retried n times, right?